Assets/Scripts/SettingsInfo.cs

```diff
@@ -0,0 +1,253 @@
+    private AudioMixer mainAudioMixer;
+    #endregion
+    #region Gameplay variables
+
+    [Header("Sensitivity Limits")]
+    public float lowerSensLimit = 0.1f;
+    public float upperSensLimit = 3f;
+
+    [Header("FOV Limits")]
+    public float lowerFOVLimit = 60f;
+    public float upperFOVLimit = 120f;
+
+    [Header("Zoom FOV Limits")]
+    public float lowerZoomFOVLimit = 20f;
+    public float upperZoomFOVLimit = 50f;
+
+    #endregion
+    public Settings settings = new()
+    {
+        sensScale = 1.0f,
+        masterVolume = 1.0f,
+        musicVolume = 1.0f,
+        sfxVolume = 1.0f,
+        qualityPresetIndex = 0,
+        resolutionPresetIndex = 0,
+        displayMode = 0,
+    };
+
+    private void Awake()
+    {
+        resolutions = Screen.resolutions.Reverse().ToArray();
+        qualityNames = QualitySettings.names;
+
+        mainAudioMixer.GetFloat(audioGroupMusic, out float musicVolume);
+        maxVolumeMusic = musicVolume;
+        mainAudioMixer.GetFloat(audioGroupSFX, out float sfxVolume);
+        maxVolumeSFX = sfxVolume;
+    }
+
+    void Start()
+    {
+        #region Singleton boilerplate
+
+        if (Singleton != null)
+        {
+            if (Singleton != this)
+            {
+                Debug.LogWarning($"There's more than one {Singleton.GetType()} in the scene!");
+                Destroy(gameObject);
+            }
+
+            return;
+        }
+
+        Singleton = this;
+
+        #endregion Singleton boilerplate
+
+        FilePath = Application.persistentDataPath + FileName;
+
+        if (!File.Exists(FilePath))
+        {
+            CreateDefaultFile();
+        }
+        LoadSettingsFile();
+        ApplyAllSettings();
+
+        DontDestroyOnLoad(gameObject);  
+    }
+    #region Save methods
+    private void LoadSettingsFile()
+    {
+        try
+        {
+            string jsonData = File.ReadAllText(FilePath);
+            settings = JsonUtility.FromJson<Settings>(jsonData);
+            Debug.Log("Settings data loaded");
+            ValidateSettings();
+        }
+        catch
+        {
+            Debug.Log("Settings file corrupted or missing");
+            CreateDefaultFile();
+        }
+    }
+
+    private void ValidateSettings()
+    {
+        // Check limits for Audio
+        Mathf.Clamp(settings.masterVolume, 0f, 1f);
+        Mathf.Clamp(settings.musicVolume, 0f, 1f);
+        Mathf.Clamp(settings.sfxVolume, 0f, 1f);
+
+        // Check limits for Graphics
+        Math.Clamp(settings.resolutionPresetIndex, 0, resolutions.Length - 1);
+        Math.Clamp(settings.qualityPresetIndex, 0, qualityNames.Length - 1);
+        Math.Clamp(settings.displayMode, 0, 3);
+
+        // Check lower limits for Gameplay
+        if (settings.sensScale <= 0f)
+            settings.sensScale = 0.01f;
+
+        if (settings.playerFOV < 20f)
+            settings.playerFOV = 20f;
+
+        if (settings.zoomFOV < 5f)
+            settings.zoomFOV = 5f;
```

Comment:
These could be replaced with `settings.zoomFOV = Mathf.Max(5f, settings.zoomFOV)` if you want :))))